stardog/admin.py

```diff
@@ -1080,8 +1080,14 @@ def __init__(self, name, client):
         self.query_name = name
         self.client = client
         self.path = "/admin/queries/stored/{}".format(name)
-        self.details = {}
-        self.__refresh()
+        """ if details is valid type and not empty set details
+            and skip refresh
+        """
+        if details and type(details) is dict:
```

Comment:
After changing the default to `None` you can update this, and try to avoid using `type` directly instead this should be:

```
if details is not None and isinstance(details, dict):
```